### ISSUE TITLE
Feature Flag description not being set

### ISSUE THREAD

**Issue by MarioPires** [Azure/azure-sdk-for-python#34771]
- **Package Name**: azure.appconfiguration
- **Package Version**: 1.5.0
- **Operating System**: Windows
- **Python Version**: 3.11

**Describe the bug**
Calling `client.set_configuration_setting` with a `FeatureFlagConfigurationSetting` with a description does not set the description.

**To Reproduce**
```
from azure.appconfiguration import (
    AzureAppConfigurationClient,
    FeatureFlagConfigurationSetting,
)
from azure.identity import DefaultAzureCredential


url = "https://my_app_config.azconfig.io"

client = AzureAppConfigurationClient(url, DefaultAzureCredential())

feature_flag_setting = FeatureFlagConfigurationSetting(
    "Test", enabled=True, description="testing sdk", label="testing"
)

client.set_configuration_setting(configuration_setting=feature_flag_setting)
```

**Expected behavior**
Description should be set with the designated value.

**Screenshots**
After running code above:
![image](https://github.com/Azure/azure-sdk-for-python/assets/10818781/8ccd23d4-1e9e-4740-96a8-3de2ecb81a7b)

**Comment by annatisch** [comment]
Thanks for the report @MarioPires - we will investigate asap!
From what I can see in the code - it seems that `description` is not being passed through to the REST API (also `display_name`):
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py#L258-L268
@YalinLi0312 - could you please investigate this? I'm not use if these fields are supposed to be passed through as high level parameters, or whether they're supposed to be embedded into `value` or `tags`? This seems to be an issue for both serialization and deserialization.

Also - I'm not sure why the rest of these keyword-only constructor params are not in the docstring?

**Comment by YalinLi0312** [comment]
* These values are sent or received via property `value`. 
* Yes, it's a bug in serialization and deserialization for `value` values, we assumed the value in schema: '{"id": ..., "enabled": ..., "conditions": {"client_filters": [...]}}' when de/serialization. 

@MarioPires  Thanks for your contact, I'll fix it in a patch soon!

**Comment by YalinLi0312** [comment]
Hi @MarioPires , we've fixed the issue in version [1.6.0b2 ](https://pypi.org/project/azure-appconfiguration/1.6.0b2/). Please try on it, thanks!
